Generator: refine the generated code for algebraic models

agarny opened the pull request:
Fixes #899.

agarny committed:
Generator: refined the generated code for algebraic models.

agarny committed:
Merge branch 'main' into issue899.

agarny committed:
Generator: refined the generated code for algebraic models.

agarny committed:
Generator: changed the version of our code generator (0.2.0 -> 0.3.0).

agarny committed:
Scripts: added a script to change the version of libCellML.

agarny committed:
Utilities: added a utility to compute the SHA-1 value of a generator profile.

hsorby commented:
This looks good to me but I would rather have the scripts in a separate repository.  The main reason being for use by the CI.

agarny commented:
> This looks good to me but I would rather have the scripts in a separate repository. The main reason being for use by the CI.

FWIW, I am not yet done with this PR. Otherwise, not sure how having the scripts in a separate repository would help, especially the CI...? Not to mention that I am currently working on an automatic way of updating the SHA-1 values for the generator profiles. (I am nearly done with that.)

agarny committed:
Some minor cleaning up.

agarny committed:
Scripts: added a way to retrieve the SHA-1 value of a C/Python profile.

agarny committed:
Some minor cleaning up.

agarny committed:
Some minor cleaning up.

agarny committed:
Some minor cleaning up.

agarny committed:
Scripts: added a script to automatically update the SHA-1 values of our C/Python profiles.

agarny committed:
Ignore default.profraw in our root directory.

agarny committed:
Make Clang-Tidy happy.

agarny committed:
Some minor cleaning up.

agarny committed:
Generator: requalify a variable-based constant equation as an algebraic equation if it depends on some external variables.

agarny committed:
Generator tests: improved our hodgkinHuxleySquidAxonModel1952WithDependentComputedConstantAsExternalVariable test.

agarny committed:
Generator: requalify a computed constant variable as an algebraic variable if it depends on some external variables.

agarny commented:
@hsorby & @nickerso: this is now ready for re-reviewing.

I have addressed the bug @nickerso spotted on Tuesday. The original implementation just didn't account for the fact that a variable-based constant could be requalified as an algebraic variable if it was to depend on some external variables. Oopsie!

I have also added some scripts to update all of our files and SHA-1 values when we want to change the version of libCellML or of the code generator.

agarny committed:
Merge branch 'main' into issue899.

agarny committed:
Merge branch 'main' into issue899

agarny committed:
Merge branch 'main' into issue899

agarny committed:
Removed the scripts.

agarny committed:
Merge branch 'main' into issue899

agarny committed:
Merge branch 'main' into issue899

nickerso committed:
Merge branch 'main' into issue899
